From e1e908801d938871ca8c6bb000061c399901818c Mon Sep 17 00:00:00 2001
From: Mariana Meireles <mariana@psychonautgirl.space>
Date: Mon, 10 May 2021 19:17:07 +0200
Subject: [PATCH] Define specific mysql versions to use my_bool or bool

---
 src/backends/mysql/session.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/backends/mysql/session.cpp b/src/backends/mysql/session.cpp
index 383248cc..9a533384 100644
--- a/src/backends/mysql/session.cpp
+++ b/src/backends/mysql/session.cpp
@@ -355,7 +355,11 @@ mysql_session_backend::mysql_session_backend(
     }
     if (reconnect_p)
     {
-        my_bool reconnect = 1;
+        #if MYSQL_VERSION_ID < 8
+            my_bool reconnect = 1;
+        #else
+            bool reconnect = 1;
+        #endif
         if (0 != mysql_options(conn_, MYSQL_OPT_RECONNECT, &reconnect))
         {
             clean_up();
-- 
2.30.2

